File: 0007-reverse-integer/0007-reverse-integer.py
class Solution:
    def reverse(self, x: int) -> int:
        neg = 0
        if x < 0:
            neg = 1
        rev = 0
        x = abs(x)

        while (x!=0):
            if rev > ((2**31-1)//10) or rev < ((2**(-31))//10):
                return 0
            digit = x % 10
            rev = rev * 10 + digit
            x = x // 10
        if neg == 1 :
            rev = -rev
        return rev
        

File: 0007-reverse-integer/0007-reverse-integer.cpp
#include <climits> // For INT_MAX and INT_MIN

class Solution {
public:
    int reverse(int x) {
        int reversed_num = 0;
        cout<<INT_MIN<<" "<<INT_MAX;
        while (x != 0) {
            int rem = x % 10;
            x /= 10;
            
            
            if (reversed_num > INT_MAX / 10 ) {
                return 0;
            }
            if (reversed_num < INT_MIN / 10 ) {
                return 0;
            }
            
            reversed_num = reversed_num * 10 + rem;
        }
        
        return reversed_num;
    }
};